lib/Migration/Version3003Date20180720162342.php

```diff
@@ -0,0 +1,55 @@
+<?php
+declare(strict_types=1);
+
+/**
+ *
+ * @copyright Copyright (c) 2018, Daniel Calviño Sánchez (danxuliu@gmail.com)
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Spreed\Migration;
+
+use Doctrine\DBAL\Types\Type;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\SimpleMigrationStep;
+use OCP\Migration\IOutput;
+
+class Version3003Date20180720162342 extends SimpleMigrationStep {
+
+	public function changeSchema(IOutput $output, \Closure $schemaClosure, array $options) {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		$table = $schema->getTable('talk_rooms');
+
+		if (!$table->hasColumn('object_type')) {
+			$table->addColumn('object_type', Type::STRING, [
+				'notnull' => false,
```

Comment:
Why this change? Were not empty strings allowed already (in fact the default value is an empty string)?

Reply:
https://github.com/nextcloud/activity/commit/34279aed14493d167f2f50bfae1797048cbaac2c

All hail to oracle, storing empty string as null which then fails.

Reply:
Urg... Thanks for the explanation, I added it to the commit message for future reference.